frontend/pages/queries/edit/components/EditQueryForm/EditQueryForm.tsx

```diff
@@ -188,6 +194,37 @@ const EditQueryForm = ({
   const [isSaveAsNewLoading, setIsSaveAsNewLoading] = useState(false);
   const [showAdvancedOptions, setShowAdvancedOptions] = useState(false);
   const [queryWasChanged, setQueryWasChanged] = useState(false);
+  const [selectedTargetType, setSelectedTargetType] = useState("");
+  const [selectedLabels, setSelectedLabels] = useState({});
+
+  useEffect(() => {
+    setSelectedTargetType(
+      storedQuery?.labels_include_any?.length && isPremiumTier
+        ? "Custom"
+        : "All hosts"
+    );
+    setSelectedLabels(
+      storedQuery?.labels_include_any?.reduce((acc, label) => {
+        return {
+          ...acc,
+          [label.name]: true,
+        };
+      }, {}) || {}
+    );
+  }, [storedQuery]);
+
+  const { data: labels, isFetching: isFetchingLabels } = useQuery<
```

Comment:
We've historically done more like this kind of thing to immediately process response data:

```
  const { data: labels, isFetching: isFetchingLabels } = useQuery<
    ILabelsSummaryResponse,
    Error,
    ILabelSummary[]
  >(["custom_labels"], labelsAPI.summary, {
    ...DEFAULT_USE_QUERY_OPTIONS,
    enabled: isPremiumTier,
    staleTime: 10000,
    select: (res) => getCustomLabels(res.labels),
  });
  ```

  Any strong opinions for `.then` etc.?

Reply:
No strong feelings, updated.